### ISSUE TITLE
Segfault with new image and rus+mrz in recent main

### ISSUE THREAD

**Issue by marcreichman-pfi** [tesseract-ocr/tesseract#4361]
### Current Behavior

With the attached image, and recent `main` (`9f17a3fd`) I receive a segfault in Release (SIGILL in Debug) with the `mrz` and `rus` langages. Both are `fast`. Rus is official and mrz is available [here](https://github.com/DoubangoTelecom/tesseractMRZ/)

Release:
```
<snip>/tesseract-fork/cmake-build-release/bin/tesseract <snip>/62553353F7495226759F9644269AA485-sample_000101.jpg - --tessdata-dir <snip>/tessdata/ -l mrz+rus
Estimating resolution as 971

Process finished with exit code 139 (interrupted by signal 11:SIGSEGV)
```

Debug:
```
Estimating resolution as 971
ELIST2_ITERATOR::add_before_stay_put:Error:Attempting to add an element with non nullptr links, to a list

Process finished with exit code 132 (interrupted by signal 4:SIGILL)
```

![new-segfault-20241125](https://github.com/user-attachments/assets/2d148458-9009-4bb2-9771-5bcfcda67ed1)

### Expected Behavior

No segfault.

### tesseract -v

```
tesseract 5.5.0-26-g9f17a
 leptonica-1.82.0
  libgif 5.1.9 : libjpeg 8d (libjpeg-turbo 2.1.1) : libpng 1.6.37 : libtiff 4.3.0 : zlib 1.2.11 : libwebp 1.2.2 : libopenjp2 2.4.0
 Found AVX
 Found SSE4.1
 Found OpenMP 201511
```

### Operating System

Ubuntu 22.04 Jammy

### Other Operating System

This is running via WSL but it does not appear related to the issue.

### uname -a

`Linux hostname 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

### Compiler

GCC 11.4

### CPU

`Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz`

### Virtualization / Containers

No explicit VM except for WSL2

### Other Information

This works in `5.5.0` tag

**Comment by egorpugin** [comment]
Now please download this file
https://github.com/tesseract-ocr/tesseract/blob/5c78037bd3d79dbb71bdccf6333a8d94b530fd19/src/ccutil/helpers.h
replace `src/ccutil/helpers.h` with it
rebuild debug build and try again.

**Comment by egorpugin** [comment]
Yes, that is what I'm testing.

**Comment by egorpugin** [comment]
It is better to keep that bunch of tickets open until the real issue(s) are fixed.

**Comment by amitdo** [comment]
Try to add a black frame to the image.

`convert  in.jpg  -bordercolor Black -border 10x10 in-black-bprder.png`

**Comment by amitdo** [comment]
Please try it and tell me if after adding the black frame you still get a segfault.

**Comment by stweil** [comment]
@marcreichman-pfi, could you please give https://github.com/tesseract-ocr/tesseract/issues/4362#issuecomment-2499107231 a try? Do you still get crashs with this code change? And similar important: does it change the OCR results?

**Comment by egorpugin** [comment]
definitely the issue should be investigated with this change
```
-      line[i] = IntCastRounded(randomizer->SignedRand(INT8_MAX));
+      line[i] = 0;
```

**Comment by egorpugin** [comment]
Yes, I mean we should track down the issue using 0 value instead of any randoms.
